From b72646f60c5b7849e87f318f2176d442901bc932 Mon Sep 17 00:00:00 2001
From: zachary <zhangzg@marvell.com>
Date: Sat, 31 Dec 2016 00:42:01 +0800
Subject: [PATCH 1236/2241] fix: spi: a3700: correct clock mode configuration

- SPI controller clock mode used to only be set when CPOL or CPHA
  bits change from 0 to 1, which is not correct, since it could
  not switch back from 1 to 0 again.
- The patch takes bits CPOL and CPHA, and configures them directly
  to SPI controller, no matter the values are 0 or 1.

Change-Id: I9f3cf4158925b0259889848a8c2d8dfd1f5717b2
Signed-off-by: zachary <zhangzg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35175
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hua Jing <jinghua@marvell.com>
---
 drivers/spi/spi-armada-3700.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/spi/spi-armada-3700.c b/drivers/spi/spi-armada-3700.c
index e8e2e23..f94a618a 100644
--- a/drivers/spi/spi-armada-3700.c
+++ b/drivers/spi/spi-armada-3700.c
@@ -256,15 +256,17 @@ static int a3700_spi_fifo_flush(struct a3700_spi *a3700_spi)
 }
 
 static void a3700_spi_mode_set(struct a3700_spi *a3700_spi,
-	unsigned int mode_bits)
+	u16 mode)
 {
 	u32 val;
 
 	val = spireg_read(a3700_spi, A3700_SPI_IF_CFG_REG);
-	if (mode_bits & SPI_CPOL)
+	val &= ~(A3700_SPI_CLK_POL | A3700_SPI_CLK_PHA);
+	if (mode & SPI_CPOL)
 		val |= A3700_SPI_CLK_POL;
-	if (mode_bits & SPI_CPHA)
+	if (mode & SPI_CPHA)
 		val |= A3700_SPI_CLK_PHA;
+
 	spireg_write(a3700_spi, A3700_SPI_IF_CFG_REG, val);
 }
 
-- 
2.7.4

